File: Stack/stack_ll.cpp
#include<iostream>
using namespace std;
int count=0;
struct node
{
    int value;
    struct node *next;
};
struct node *top;

class stack
{
    public:
        int x;
        void push()
        {
            cout<<"Enter the number to insert"<<endl;
            cin>>x;
            struct node *temp;
            temp=new node;
            temp->value=x;
            temp->next=NULL;
            if(top==NULL)
            top=temp;
            else
            {
                temp->next=top;
                top=temp;
            }
            count++;
            cout<<x<<" is inserted in stack in position "<<count<<endl;
        }
        void pop()
        {
            struct node *temp;
            temp=new node;
            if(top==NULL)
            cout<<"Stack is empty , error in performing pop operation";
            else
            {
            temp=top;
            top=temp->next;
            count--;
            cout<<temp->value<<" is removed from stack"<<endl;
            delete temp;
            }
        }
        void show_top()
        {
            if(top==NULL)
            cout<<"Stack is empty, top is not assigned\n";
            else
            cout<<"==========     top-> "<<top->value<<"     ==========";
        }
        void show()
        {
            struct node *temp;
            temp=new node;
             cout<<"--------------------------"<<endl;
            if(top==NULL)
            cout<<"Stack is empty";
            else
            {
            cout<<"Elements in stack"<<endl;
            for(temp=top;temp!=NULL;temp=temp->next)
            cout<<temp->value<<" ";
            }
            cout<<"\n--------------------------\n";
        }
};

int main()
{
    stack s1;
    int c=1,ch;
    while(c)
    {
        cout<<"1.Push\n2.Pop\n3.Show Top\n4.display Stack\n5.No of elements in stack"<<endl;
        cin>>ch;
        switch(ch)
        {
            case 1:
            s1.push();
            break;
            case 2:
            s1.pop();
            break;
            case 3:
            s1.show_top();
            break;
            case 4:
            s1.show();
            break;
            case 5:
            cout<<"Elements in stack = "<<count;
            break;
            default:
            cout<<"\nPlease select valid operation"<<endl;
            break;
        }
        cout<<"\n-----------\n Do you want to continue? (1 to continue /// 0 to exit)"<<endl;
        cin>>c;
    }
}

File: Stack/stack_ll.java
import java.util.*;

class node
{
    int value;
    node next;

    node()
    {
        next=null;
    }
}

class stack
{
    public node top;
    public int count=0;
    Scanner sc=new Scanner(System.in);
    stack()
    {
        this.top=null;
    }
    public void push()
    {
        System.out.println("Enter the element to insert in stack");
        int element=sc.nextInt();
       
        node temp=new node();
        temp.value=element;

        if(top==null)
        {
            top=temp;
        }
        else
        {
            temp.next=top;
            top=temp;
        }
        count++;
        System.out.println(element+" is inserted in stack at position "+count);
    }
    public int pop()
    {
        if(top==null)
        return -1;
        else
        {
            int element=top.value;
            top=top.next;
            count--;
            return element;
        }
    }
    public void showTop()
    {
        if(top==null)
        System.out.println("Stack is empty, top is not assigned");
        else
        System.out.println("==========     top-> "+top.value+"     ==========");
    }
    public void show()
    {
        node temp=new node();
        System.out.println("--------------------------");
        if (top==null)
        System.out.println("Stack is empty");
        else
        {
        System.out.println("Elements in Stack -");
        for(temp=top;temp!=null;temp=temp.next)
            System.out.println(temp.value+" ");
        }
        System.out.println("--------------------------");
    }
}

public class stack_ll
{
    public static void main(String args[])
    {
        try(Scanner sc=new Scanner(System.in))
        {
        stack stack=new stack();
        int control=1,choice;
        while(control==1)
        {
            System.out.println("1.Push\n2.Pop\n3.Show Top\n4.Display Stack\n5.No of elemnts in stack");
            choice=sc.nextInt();
            switch(choice)
            {
                case 1:
                    stack.push();
                    break;
                case 2:
                    int res=stack.pop();
                    if(res==-1)
                        System.out.println("Stack is empty , error in performing pop operation");
                    else
                        System.out.println(res+" is removed from stack");
                    break;
                case 3:
                    stack.showTop();
                    break;
                case 4:
                    stack.show();
                    break;
                case 5:
                    System.out.println("Elements in stack = "+stack.count);
                    break;
                default:
                    System.out.println("\nPlease select valid operation");
            }
            System.out.println("\n------------------\nDo you want to continue? (1 to continue /// 0 to exit)");
            control=sc.nextInt();
        }
        }

    }
}